cpp/daal/include/services/internal/sycl/level_zero_module_sycl.h

```diff
@@ -68,8 +68,14 @@ class ZeModule : public Base
 public:
     static SharedPtr<ZeModule> create(cl::sycl::queue & deviceQueue, size_t binarySize, const uint8_t * pBinary, Status & status)
     {
-        const auto ptr = new ZeModule(deviceQueue, binarySize, pBinary, status);
-        if (!ptr) status |= ErrorMemoryAllocationFailed;
+        auto ptr = new ZeModule(deviceQueue, binarySize, pBinary, status);
+        if (!status)
+        {
+            if (ptr) delete ptr;
+            ptr = nullptr;
+        }
+        else if (!ptr)
```

Comment:
Maybe it makes sense to check "ptr" before "status"? Just in order to ensure that ErrorMemoryAllocationFailed is added in any case.

Reply:
It is more probable to get incorrect status than nullptr and provides more info, therefore i set it first.